docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,540 @@
+
+
+Requirements
+------------
+
+Requirement: Ability to perform vertical Lagrangian-remapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+The core of the proposed development is the ability to perform vertical 
+Lagrangian-remapping (VLR). When VLR is performed, the momentum, volume and 
+tracer conservation equations are first solved in a Lagrangian sense; that is, 
+layer thicknesses evolve such that there is no vertical transport across layer 
+interfaces. A target grid is then specified. For this stage of development, 
+we only require that the algorithm be able to use the existing grid 
+specifications. However, the implementation should be general enough that 
+a different grid could be used for remapping. This regridding step is followed 
+by a conservative remapping of velocity and scalars to the target grid. 
+
+Requirement: Grid is sufficiently conditioned for accuracy and stability
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+A severely deformed grid can lead to issues with the accuracy and stability of 
+the numerical methods used for the dynamics and thermodynamics. If layers are 
+allowed to become very thin they can also degrade the accuracy and stability of 
+the solution. Thus, a requirement of the VLR implementation is that there are 
+safeguards to prevent layers from becoming too thin or deformed. 
+
+Requirement: Ability to specify remapping method and its options
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+There are several choices to make for the remapping operation to balance accuracy, 
+stability and computational cost. To the extent that it is feasible, the user 
+should have the ability to control the order of accuracy of the remapping method,
+[Darren, other options here?] and other remapping options (e.g., via namelist options).
+
+Requirement: Support for existing time stepping schemes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/04
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Vertical Lagrangian remapping should be supported (eventually) in both the RK4 
+and the split-explicit time stepping schemes currently implemented in MPAS-Ocean. 
+
+Requirement: not climate changing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/12/15
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+For the same target grid, time-stepping scheme and problem, the solution for 
+layer thicknesses and key prognostic variables using VLR is within a non-
+climate-changing threshold value [presumably greater than machine precision?] 
```

Comment:
@vanroekel Is there an agreed-upon threshold for allowable differences in the solution?

Reply:
@cbegeman if the changes are nonBFB I think we have to go with expert judgement on if we are okay with the changes.  It would be very interesting to try and use the capability from Salil Mahajan to test for an unchanged climate.  I'm not comfortable putting a specific threshold on this requirement.